### PR TITLE
config: Add private_session config

### PR DESCRIPTION
**Pull request by chigley**
Hello. Thanks so much for Mopidy and the Spotify extension!

I occasionally find myself in situations in which I'd prefer my Spotify session to be private. This PR adds a `private_session` config value which can be used to do just that :smile:

### CODE REVIEW

File: tests/test_backend.py [hunk 1]
```diff
@@ -77,6 +77,17 @@ def test_on_start_configures_volume_normalization(spotify_mock, config):
     volume_normalization_mock.assert_called_once_with(False)
 
 
+def test_on_start_configures_private_session(spotify_mock, config):
+    session = spotify_mock.Session.return_value
+    private_session_mock = mock.PropertyMock()
+    type(session).private_session = private_session_mock
```
Comment (jodal) [inline tests/test_backend.py]: The test fails, probably due to this line adding the mock to the wrong object. It's `session.social.private_session`, not `session.private_session`.

Comment (chigley) [inline tests/test_backend.py]: Thanks for the fast response! Yeah, my bad, fixed in dfa3d3b272efe3de720bd0333a3972505eb9ac1d and the CI build is passing now.

File: README.rst [hunk 2]
```diff
@@ -126,6 +126,9 @@ The following configuration values are available:
   domains to get toplists for. Defaults to a long list of countries which
   Spotify is available in.
 
+- ``spotify/private_session``: Whether to use a private Spotify session.
```
Comment (jodal) [inline README.rst]: I'd like this to describe what it does too, maybe something like:

> Turn on private session to disable sharing of played tracks with friends through the Spotify activity feed, Last.fm scrobbling, and Facebook. This only affects social sharing done by Spotify, not by other Mopidy extensions.

**Comment by jodal** [comment]
Thanks for a lovely pull request: a single specific change, with tests, docs, and perfect formatting. :-)

**Comment by chigley** [comment]
Heh, just a shame about the silly failing test really :wink: Thanks for merging!

**Pull request merged**